libs/api/libbpf_link.cpp

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+#include "api_internal.h"
+#pragma warning(push)
+#pragma warning(disable : 4200)
+#include "libbpf.h"
+#pragma warning(pop)
+#include "libbpf_internal.h"
+
+int
+bpf_link__pin(struct bpf_link* link, const char* path)
+{
+    int err;
+
+    if (link->pin_path)
+        return -EBUSY;
+
+    link->pin_path = strdup(path);
+    if (!link->pin_path)
+        return -ENOMEM;
+
+    if (ebpf_object_pin(link->link_fd, link->pin_path)) {
+        err = -errno;
```

Comment:
where is errno set?  Right now, errno is always set in libbpf_* and never in ebpf_api.cpp.
Is it set by some underlying system call that ebpf_api.cpp calls?  If so, then this is ok as is.